File: C1/reinforce/R1-3.py
# Write a short python function minmax that thakes a sequence of one or more numbers, and returns the result in the form of a tuple of lenght two. Dont use built in functions minmax.

def minmax(data):

    i = 0
    max = 0
    min = 1
    
    for i in range(len(data)):
        if (data[i] > max):
            max = data[i]
        
        if (data[i] < min):
            min = data[i]
       
    return f"({min}, {max})"

# Testing

print(minmax([1, 2, 3]))

File: C1/reinforce_in_C/R1-3.c
#include <stdio.h>

typedef struct{

	float min;
	float max;
} MinMax;

MinMax minmax(float data[], int n){

	MinMax result;
	result.min = data[0];
	result.max = data[0];

	for (int i = 1; i < n; i++){
		if(data[i] > result.max){
			result.max = data[i];
		}
		if(data[i] < result.min){

			result.min = data[i];
		}

	}

	return result;
}

int main() {
    float numbers[] = {1, 2, 3};
    int size = sizeof(numbers) / sizeof(numbers[0]);
    
    MinMax result = minmax(numbers, size);
    
    printf("(%.2f, %.2f)\n", result.min, result.max);
    
    return 0;
}
